File: mobile-engage/src/main/java/com/emarsys/mobileengage/notification/command/OpenExternalUrlCommand.java
package com.emarsys.mobileengage.notification.command;

import android.content.Context;
import android.content.Intent;

import com.emarsys.core.util.Assert;

public class OpenExternalUrlCommand implements Runnable {

    private final Context context;
    private final Intent intent;

    public OpenExternalUrlCommand(Intent intent, Context context) {
        Assert.notNull(intent, "Intent must not be null!");
        Assert.notNull(context, "Context must not be null!");
        this.intent = intent;
        this.context = context;
    }

    @Override
    public void run() {
        context.startActivity(intent);
    }

    public Context getContext() {
        return context;
    }

    public Intent getIntent() {
        return intent;
    }
}

File: mobile-engage/src/main/java/com/emarsys/mobileengage/notification/command/OpenExternalUrlCommand.kt
package com.emarsys.mobileengage.notification.command

import android.content.Context
import android.content.Intent
import com.emarsys.core.Mockable
import com.emarsys.core.util.SystemUtils
import com.emarsys.core.util.log.Logger
import com.emarsys.core.util.log.entry.StatusLog

@Mockable
class OpenExternalUrlCommand(val intent: Intent, val context: Context) : Runnable {
    override fun run() {
        try {
            context.startActivity(intent)
        } catch (exception: Exception) {
            Logger.debug(StatusLog(OpenExternalUrlCommand::class.java, SystemUtils.getCallerMethodName(), mapOf(
                    "intent" to intent,
                    "exception" to exception
            )))
        }
    }
}